subsys/shell/modules/device_service.c

```diff
@@ -39,7 +39,7 @@ static bool device_get_config_level(const struct shell *shell, int level)
 	bool devices = false;
 
 	for (dev = levels[level]; dev < levels[level+1]; dev++) {
-		if (dev->driver_api != NULL) {
+		if (dev->driver_context->init_ok) {
```

Comment:
hm, if we stick to the goal of this RFC: const-ifying struct device instances, then such driver context is overkill.

Better having an array of atomic bits then, like PM's busy bit array. Less changes to reach the goal.